### DISCUSSION TITLE
Multiply PAS by throttle?

### DISCUSSION THREAD

**Discussion by kivig2** [danielnilsson9/bbs-fw#22] [General]
Although I'm afraid to try it yet, your firmware seems to be the most progressive of what can be found feature-wise. Hope it will make it to a release.
One thing though nobody is trying for some weird reason is throttle controlled pedal assist.
For one - in many countries it is illegal to have a motor that drives when not pedaling, meaning classic throttle is illegal. But more importantly - it is so much more satisfying to have a fluid control as opposed to always over/under-powered, jerky/laggy assist that needs braking or other clumsy ways of temporary disabling when you don't want it to kick in.
Don't yet know the reasons for not being implemented, but it seems so logical to just multiply the existing assist by existing throttle in [0..1] range.
I'll try to read up on code, but I'd be curious to know if/why could it be not possible/feasible?

**Comment by danielnilsson9** [comment]
I should be a pretty simple thing to do really. See apply_pas() in app.c and check where it is called in app_proces(). there you have access to the throttle reading.

But there is an issue with unreliable throttle reading so I do not recommend using that at the moment.

**Comment by danielnilsson9** [comment]
This was implemented and is now part of the firmware. The assist mode is called "Variable PAS" in the config tool and documentation.

**Comment by dav0000000** [comment]
This is theoretically "activate the throttle only if pedalling” ???

That would basically just need "if (pas_is_pedaling_forwards()) {...}" added to the apply_throttle() function somewhere.

Not sure that would get around any laws which specifically talk about throttle if that is your purpose.

Dave

On 2 May 2022, at 2:40 am, kivig2 ***@***.***> wrote:


Although I'm afraid to try it yet, your firmware seems to be the most progressive of what can be found feature-wise. Hope it will make it to a release.
One thing though nobody is trying for some weird reason is throttle controlled pedal assist.
For one - in many countries it is illegal to have a motor that drives when not pedaling, meaning classic throttle is illegal. But more importantly - it is so much more satisfying to have a fluid control as opposed to always over/under-powered, jerky/laggy assist that needs braking or other clumsy ways of temporary disabling when you don't want it to kick in.
Don't yet know the reasons for not being implemented, but it seems so logical to just multiply the existing assist by existing throttle in [0..1] range.
I'll try to read up on code, but I'd be curious to know if/why could it be not possible/feasible?

**Reply by kivig2** [reply]
"Not sure that would get around any laws which specifically talk about throttle if that is your purpose."
The law states that the bicycle can be fitted with a motor that is limited in speed and disengages when the pedals aren't turned. It doesn't forbid limiting the power additionally via throttle. Don't know how many countries define it like that.
